starfish/pipeline/features/spots/detector/local_max_peak_finder.py

```diff
@@ -0,0 +1,143 @@
+
+        spot_diameter : odd integer or tuple of odd integers.
+            This may be a single number or a tuple giving the feature’s extent in each dimension, useful when the
+            dimensions do not have equal resolution (e.g. confocal microscopy). The tuple order is the same as the
+            image shape, conventionally (z, y, x) or (y, x). The number(s) must be odd integers. When in doubt, round
+            up.
+        min_mass : float, optional
+            The minimum integrated brightness. This is a crucial parameter for eliminating spurious features.
+            Recommended minimum values are 100 for integer images and 1 for float images. Defaults to 0 (no filtering).
+        max_size : float
+            maximum radius-of-gyration of brightness, default None
+        separation : float or tuple
+            Minimum separtion between features. Default is diameter + 1. May be a tuple, see diameter for details.
+        percentile : float
+            Features must have a peak brighter than pixels in this percentile. This helps eliminate spurious peaks.
+        noise_size : float or tuple
+            Width of Gaussian blurring kernel, in pixels Default is 1. May be a tuple, see diameter for details.
+        smoothing_size : float or tuple
+            The size of the sides of the square kernel used in boxcar (rolling average) smoothing, in pixels Default
+            is diameter. May be a tuple, making the kernel rectangular.
+        threshold : float
+            Clip bandpass result below this value. Thresholding is done on the already background-subtracted image.
+            By default, 1 for integer images and 1/255 for float images.
+        preprocess : boolean
+            Set to False to turn off bandpass preprocessing.
+        max_iterations : integer
+            max number of loops to refine the center of mass, default 10
+        is_volume : bool
+            if True, run the algorithm on 3d volumes of the provided stack
+        verbose : bool
+            If True, report the percentage completed (default = False) during processing
+
+
+        See Also
+        --------
+        trackpy locate: http://soft-matter.github.io/trackpy/dev/generated/trackpy.locate.html
+
+        """
+
+        self.diameter = spot_diameter
+        self.minmass = min_mass
+        self.maxsize = max_size
+        self.separation = separation
+        self.noise_size = noise_size
+        self.smoothing_size = smoothing_size
+        self.percentile = percentile
+        self.threshold = threshold
+        self.preprocess = preprocess
+        self.is_volume = is_volume
+        self.verbose = verbose
+
+    # # TODO ambrosejcarr: make this generalize to smFISH methods
+    # def encode(self, spot_attributes: SpotAttributes):
+    #     spot_table = spot_attributes.data
+    #     spot_table['barcode_index'] = np.ones(spot_table.shape[0])
+
+    def find_attributes(self, image):
+        attributes = locate(
```

Comment:
does this require image to be a specific format?

Reply:
It interacts with the parameters, but no, I don't think so. I'll add a docstring that gives some details.